test/cloudtest/test_secrets.py

```diff
@@ -103,6 +103,9 @@ def test_missing_secret(mz: MaterializeApplication) -> None:
           $ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
           ALTER SYSTEM SET enable_connection_validation_syntax = true
 
+          $ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+          ALTER SYSTEM SET enable_secrets_caching = false
+
```

Comment:
@alex-hunt-materialize I want to specifically this change out. Here we're testing what happens if a secret goes missing from Kubernetes. We need to disable the cache because otherwise we'd read the secret from the cache (until it times out) so the test would fail

Reply:
I think this problem goes away if we separate out the `SecretsController` and the `CachingSecretsReader`. IMO, the `CREATE SOURCE` command should use the non-caching secrets controller to always get the latest version of a secret, and it'll just be active webhooks that use the caching secrets reader.